### ISSUE TITLE
sweep all wbtc from contract sent there by mistake

### ISSUE THREAD

**Issue by gosuto-inzasheru** [WrappedBTC/DAO#17]
quite some funds were sent to the wbtc contract directly by mistake over the years.

luckily the contract has a function (`reclaimToken`) which sweeps a particular erc20 back to `owner`. the controller would need to call that function, forward all those tokens to the newly deployed safe msig and from there they can be distributed amongst users.

naive airdrop csv file (ie return to sender), compatible with safe's csv airdrop app:
```
token_type,token_address,receiver,amount,id
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0x53ddfb32cb24b0712110455c77baf127a30e413a,0.03024841,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0x29bc2188262477707becab36dcb3b2a466ec076f,0.00970413,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0xbe39557e41425341711e594d51d130c765791f83,0.001,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0xa1a32679bdb04e8f969429c396547a69ca1c353b,0.00015498,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0x111b46246ba9bf46ef15dca39ab7775897271951,0.05405733,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0xd1669ac6044269b59fa12c5822439f609ca54f41,0.0000037,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0x382126c0849d6f62ebc49c3732394ecc482f0767,8.793148,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0xfbb1b73c4f0bda4f67dca266ce6ef42f520fbb98,0.00134994,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0x46340b20830761efd32832a74d7169b29feb9758,0.006868,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0x617e1eeddc9953beabdf21e03b8ecc1cfdc39149,0.07854583,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0xd8a4aa0ca3e0f47bd6bf2547b5ab8809526cefd0,0.0078,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0xe5730adb126dcc20dd00c866c57f26dad3cf92d4,0.00457314,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0x82e203d3453714b1824e63e82cf9a56d09c869d0,0.00059668,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0x44b465323f4fcaa88c2fa46df1f6b737b3bc3a0f,0.01699603,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0x7209e50bd2311cba873a95117038575a1d65436d,0.00055,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0x00c67d9d6d3d13b42a87424e145826c467cccd84,0.21106567,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0xaf48a699bc0a049eb64c57c61feb427b28cd97f4,0.17284604,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0xa6d146c14f213fbf172196891cef0d523a8652eb,0.01098592,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0x0093e5f2a850268c0ca3093c7ea53731296487eb,0.00152611,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0x1d3d346223c905dff1a5343e89fe37a59ddc3a95,0.0019569,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0x0753cfbc797abfce05abaacbb1e6ae032feb5f1d,0.01,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0x3a54ec3dee0a6f633e2666e7ba3b59b7839f969d,0.0001,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0x6cc5f688a315f3dc28a7781717a9a798a59fda7b,0.001,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0xa6aaa94ab6f9525818ba6b1a890f65aa5775b72b,0.00058677,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0x3cd751e6b0078be393132286c442345e5dc49699,0.03,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0x7f149ca6b8eca0575a690f875cb3cbce27d157fe,0.00069053,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0x785b70b908e1458bdae818c03d7b8e6688b025e6,0.005,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0x4da054fb5eef39b2b6021458c5afb95317c9a3a1,0.00829879,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0x695d84e20613f30dcd2ce5c9428c9ed37a0137d2,0.0056518,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0x3cd751e6b0078be393132286c442345e5dc49699,0.00086,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0x871d5c8ff2349fba48da9aee2530b9688538ef9b,0.01,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0x990cadcc315896f623bdc945e862f30c46391179,0.001,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0x3d7f67558d142389fd671c227924afe788249174,0.041983,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0xe4d1af0176aee557e6e79f0c3ad6fc37f1f9067f,0.00655,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0xeb40d71b26e3df076883da87345c05a3b880acda,0.007,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0x50238ae3d35d86ab1d2da80ba20224a9ecf06087,0.00131676,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0xd066e19c42d7bbe1cdda612e72e1589289971f65,0.032735,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0x318ed7fe131403e38ea6539e6e56583b9ec90e86,0.00186063,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0xf5174b5172942455961d442ba09e33d8fa708852,0.00139142,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0xbc13c5c64d1bb4f215387199309a61e7bfeadf42,0.00316434,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0xaa5194f0c753f813823cb5f1af5e3c12b6e07526,0.05,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0x3a99aadc22981fac57c2ea0cea84766e33b577ef,0.006,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0xb77012f205b9d5a39cdb97a529d7687be7a5d0b8,0.00937877,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0x3cd751e6b0078be393132286c442345e5dc49699,0.0006,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0xd066e19c42d7bbe1cdda612e72e1589289971f65,0.00002968,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0xd066e19c42d7bbe1cdda612e72e1589289971f65,0.00002978,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0x3cd751e6b0078be393132286c442345e5dc49699,0.00157768,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0x2e86c1e969f3c121ed865cf4d7879744c9c353df,0.5283369,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0xf83bec079b6cfba4ae0768e3137251b6ee41d8c8,0.1,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0x218d04e0abf0e23cf68dac840edc492dca214f69,0.99867802,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0x3cd751e6b0078be393132286c442345e5dc49699,0.00502139,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0x13b2850907fd72a53b4adcad0c95b7c51035b63f,0.0006733,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0xcfc0f98f30742b6d880f90155d4ebb885e55ab33,0.02063959,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0xd2d0b4a6ffdca569eefe045eaf99e53b2efe6634,0.001,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0x6cc5f688a315f3dc28a7781717a9a798a59fda7b,0.0001,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0x30a11f53bae61c1c9b873ce0adab61af452b9481,0.00283022,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0x56b25a3e6e720341f6e6ee99690d2df04fb624f6,0.00165338,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0x776c18f54a08cb2dcb26f22d1c1f899a8581799b,0.00000256,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0xeed86b90448c371eab47b7f16e294297c27e4f51,0.00023156,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0xa9d1e08c7793af67e9d92fe308d5697fb81d3e43,0.01162013,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0xa9d1e08c7793af67e9d92fe308d5697fb81d3e43,0.04489517,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0xa9d1e08c7793af67e9d92fe308d5697fb81d3e43,0.00153818,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0xcc850abe97204a34b2f8b701cec7081ab666fa2c,0.04512673,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0xcc850abe97204a34b2f8b701cec7081ab666fa2c,0.0011839,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0x586724b895ea7e866fb743a887a564bf61490f38,0.0000149,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0xcc850abe97204a34b2f8b701cec7081ab666fa2c,0.02447667,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0xcc850abe97204a34b2f8b701cec7081ab666fa2c,0.00941,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0xa216390ac9bd6c8995b8c6b452c5adc197fe5192,0.00227003,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0xa9d1e08c7793af67e9d92fe308d5697fb81d3e43,0.00294067,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0xfab576ff46bd27b095a4eee4a293ecb0c41d5a85,0.9,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0xfab576ff46bd27b095a4eee4a293ecb0c41d5a85,0.9,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0xfab576ff46bd27b095a4eee4a293ecb0c41d5a85,0.9,
```
note: some of these destinations are cex wallets (coinbase, okx, hoo, crypto.com, bittrex, coinlist) which we might want to avoid sending to (in order to prevent funds getting stuck on the other side)... that being said, it is probably impossible to determine for all destinations whether it is still alive/accessible/in use etc.

**Comment by ravz** [comment]
thanks @gosuto-inzasheru any ETA on when you would do the sweep?

**Comment by gosuto-inzasheru** [comment]
- removed all dust entries from the csv (<0.002 btc)
- removed all cex wallets as tagged by etherscan
- diff: https://www.diffchecker.com/pT6t6EvY/
- now totaling `14.01750036` btc
```
token_type,token_address,receiver,amount,id
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0x53ddfb32cb24b0712110455c77baf127a30e413a,0.03024841,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0x29bc2188262477707becab36dcb3b2a466ec076f,0.00970413,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0x111b46246ba9bf46ef15dca39ab7775897271951,0.05405733,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0x382126c0849d6f62ebc49c3732394ecc482f0767,8.793148,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0x617e1eeddc9953beabdf21e03b8ecc1cfdc39149,0.07854583,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0xd8a4aa0ca3e0f47bd6bf2547b5ab8809526cefd0,0.0078,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0xe5730adb126dcc20dd00c866c57f26dad3cf92d4,0.00457314,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0x44b465323f4fcaa88c2fa46df1f6b737b3bc3a0f,0.01699603,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0x00c67d9d6d3d13b42a87424e145826c467cccd84,0.21106567,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0xaf48a699bc0a049eb64c57c61feb427b28cd97f4,0.17284604,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0xa6d146c14f213fbf172196891cef0d523a8652eb,0.01098592,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0x0753cfbc797abfce05abaacbb1e6ae032feb5f1d,0.01,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0x785b70b908e1458bdae818c03d7b8e6688b025e6,0.005,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0x4da054fb5eef39b2b6021458c5afb95317c9a3a1,0.00829879,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0x695d84e20613f30dcd2ce5c9428c9ed37a0137d2,0.0056518,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0x871d5c8ff2349fba48da9aee2530b9688538ef9b,0.01,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0x3d7f67558d142389fd671c227924afe788249174,0.041983,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0xe4d1af0176aee557e6e79f0c3ad6fc37f1f9067f,0.00655,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0xeb40d71b26e3df076883da87345c05a3b880acda,0.007,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0xd066e19c42d7bbe1cdda612e72e1589289971f65,0.032735,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0xbc13c5c64d1bb4f215387199309a61e7bfeadf42,0.00316434,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0xaa5194f0c753f813823cb5f1af5e3c12b6e07526,0.05,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0x3a99aadc22981fac57c2ea0cea84766e33b577ef,0.006,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0xb77012f205b9d5a39cdb97a529d7687be7a5d0b8,0.00937877,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0x2e86c1e969f3c121ed865cf4d7879744c9c353df,0.5283369,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0xf83bec079b6cfba4ae0768e3137251b6ee41d8c8,0.1,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0x218d04e0abf0e23cf68dac840edc492dca214f69,0.99867802,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0xcfc0f98f30742b6d880f90155d4ebb885e55ab33,0.02063959,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0x30a11f53bae61c1c9b873ce0adab61af452b9481,0.00283022,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0xcc850abe97204a34b2f8b701cec7081ab666fa2c,0.04512673,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0xcc850abe97204a34b2f8b701cec7081ab666fa2c,0.02447667,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0xcc850abe97204a34b2f8b701cec7081ab666fa2c,0.00941,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0xa216390ac9bd6c8995b8c6b452c5adc197fe5192,0.00227003,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0xfab576ff46bd27b095a4eee4a293ecb0c41d5a85,0.9,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0xfab576ff46bd27b095a4eee4a293ecb0c41d5a85,0.9,
erc20,0x2260fac5e5542a773aa44fbcfedf7c193bc2c599,0xfab576ff46bd27b095a4eee4a293ecb0c41d5a85,0.9,
```

**Comment by tranvictor** [comment]
I inited the multisig tx to reclaim all WBTC from WBTC token contract to its owner here:

0x24abcb53aff77c7e32fed1b086ecec696d6d055a7b522692d2e19e1e27ac5e1d

<img width="650" alt="image" src="https://user-images.githubusercontent.com/5132552/230714453-ae75667e-a32e-4bb5-859b-50efeca91928.png">

After everyone approved and executed this tx, all WBTC on WBTC contract will be transferred to 0xca06411bd7a7296d7dbdd0050dfc846e95febeb7 (this is WBTC token controller contract)

```
In order to approve it:
Call contract `0xB33f8879d4608711cEBb623F293F8Da13B8A37c5.confirmTransaction(0x1)`
```

Next, we need to reclaim it from the token controller contract to its owner (WBTC dao multisig). I also init that tx here:

0x9129baedf5a6e209d1f68fec0227dffbeeefcb27b8a67516eb36661f2e47afe5

<img width="604" alt="image" src="https://user-images.githubusercontent.com/5132552/230714583-70ecc222-72f7-45b9-ba14-80e7367a9f3c.png">

After everyone approved and executed this tx, all WBTC on the controller (that was supposed to transferred after the first msig tx was executed) will be transferred to WBTC dao multisig at 0xB33f8879d4608711cEBb623F293F8Da13B8A37c5

```
In order to approve it:
Call contract `0xB33f8879d4608711cEBb623F293F8Da13B8A37c5.confirmTransaction(0x2)`
```

**Comment by ravz** [comment]
hi, who are we waiting on ?

**Comment by Smotheredhope80** [comment]
What about those that sent tokens to the contract from CEX's?

**Comment by gosuto-inzasheru** [comment]
> What about those that sent tokens to the contract from CEX's?

it is very difficult if not impossible to verify their identity and send it back, so for now cex addresses have been excluded from the airdrop csv

**Comment by tranvictor** [comment]
We have successfully transferred the WBTC to the gnosis safe msig, ready to distribute guys :) Will coordinate the distribution in the next few days.

**Comment by ravz** [comment]
hi @tranvictor thanks for this, I hope we can finally get to an end result of people having their funds returned.
Would it be possible to airdrop mine to a different address? I can prove ownership of wallet again if needed.

**Comment by tranvictor** [comment]
Hi @ravz, since it is a lot of coordination effort, although it is possible, i still strongly suggest to avoid it and still receive the fund in your original address.

**Comment by ravz** [comment]
@tranvictor ok that's fine, could you provide an estimate for when the transaction will occur? I need to transfer it out straight away (it's an old unused wallet and I dont know if its been compromised)

**Comment by tranvictor** [comment]
In this comment (https://github.com/WrappedBTC/DAO/issues/17#issuecomment-1595789722) i mistakenly said the fund was moved to the msig waiting to distribute.

The correct progress is we moved the fund to the big dao msig and since there will be many transfer, we need to move the fund to a safe gnosis msig then do another tx to distribute to everyone at once. So 2 more txs. First is to move to safe gnosis msig, Second is to distribute.

Bitgo is about to init the first one, it will take at least a week to be done since many approvals are needed. Second one will be inited right after the first one is done and it only needs 3 approvals, would take only 1-2 days IMO. So i would guess 10 days from now.
